cli/src/pcluster/config/imagebuilder_config.py

```diff
@@ -0,0 +1,149 @@
+from typing import Dict, List
+
+from pcluster.config.cluster_config import Config
+from pcluster.validators.ec2 import BaseAMIValidator
+
+
+class ImageBuilderVolumeConfig(Config):
+    """Represent the volume configuration for the ImageBuilder."""
+
+    def __init__(
+        self, size: int = 25, delete_on_termination: bool = True, encrypted: bool = False, kms_key_id: str = None
+    ):
+        super().__init__()
+        self.size = size
+        self.delete_on_termination = delete_on_termination
+        self.encrypted = encrypted
+        self.kms_key_id = kms_key_id
+        # TODO: add validator
+        self._validators = []
+
+
+class ImageBuilderComponentsConfig(Config):
+    """Represent the components configuration for the ImageBuilder."""
+
+    def __init__(self, component_arns: List[str], component_docs: Dict[str, str] = None):
+        super().__init__()
+        self.component_arns = component_arns
+        self.component_docs = component_docs
+        # TODO: add validator
+        self._validators = []
+
+
+class ImageBuilderBucketLogConfig(Config):
+    """Represent the bucket log configuration for the ImageBuilder."""
+
+    def __init__(
+        self,
+        retention_policy: str = "Retain",
+        bucket_name: str = None,  # TODO: auto generate if not assigned
+        bucket_key_prefix: str = None,  # TODO: auto generate if not assigned
+    ):
+        super().__init__()
+        self.bucket_name = bucket_name
+        self.bucket_key_prefix = bucket_key_prefix
+        self.retention_policy = retention_policy
+        # TODO: add validator
+        self._validators = []
+
+
+class ImageBuilderChefCookbookConfig(Config):
+    """Represent the chef cookbook configuration for the ImageBuilder."""
+
+    def __init__(self, url: str, json: str):
+        super().__init__()
+        self.url = url
+        self.json = json
+        # TODO: add validator
+        self._validators = []
```

Comment:
you don't need to inizialize the validators array because it's already done in the parent (the Config class), you only need to register the new validators.